test/markup/handlebars/built-ins.expect.txt

```diff
@@ -0,0 +1,12 @@
+<span class="xml"></span><span class="hljs-template-tag">{{#<span class="hljs-name"><span class="hljs-builtin-name">if</span></span> test}}</span><span class="xml">yes</span><span class="hljs-template-tag">{{/<span class="hljs-name"><span class="hljs-builtin-name">if</span></span>}}</span><span class="xml">
```

Comment:
Hmmm, any idea why the `<span class="xml"></span>`  that starts all of these?  Perhaps a bug in the sublanguage parsing?

Reply:
so far, I have only written the language after try-and-error, using my tests and looking at the result in my `testbench.html`.

The spans where there from the start, so I haven't looked at where they come from...

Reply:
Ok, Twig is the same way (it has existing markup tests), so this is just "how it works" now I guess, though it's strange... also not sure the value of raw text being supported by xml class just because it bounces in and out of that sublanguage... but oh well. :-)  

An issue for another time and place.

Reply:
I think it is the sublanguage processing:
 
*In [line 583](https://github.com/highlightjs/highlight.js/blob/master/src/highlight.js#L583), `match.index` is 0, because the token `{{{` is found directly at the beginning of the input.
*In [line 492](https://github.com/highlightjs/highlight.js/blob/master/src/highlight.js#L492), the `mode_buffer` is and remains empty because of that.
*[highlight()](https://github.com/highlightjs/highlight.js/blob/master/src/highlight.js#L464) will then be called with an empty string as value.
* buildSpan will then be called with an empty value.

I think `buildSpan` could also return and empty string instead of an empty span, if the input is empty, but I don't know all the effects, of course.

Reply:
Would you mind pasting all that into a new issue perhaps? :-)

Reply:
Adding 

```
if (!leaveOpen && insideSpan==='') {
        return ''
}
```

to the top of `buildSpan` will cause the `twig`-tests to fail as well, but the only change I see in the result is a disappearing empty `<span class="xml">`-tag. I can create another PR for that.

Reply:
I'd be happy to review it.  Need to look into if there are any deeper implications.  I'd tend to think that empty tags don't have much value though off the top of my head.

Reply:
There are other tests failing, that I haven't noticed before.